### PR TITLE
Fix arange, add floordiv, fix Fill op

### PR DESCRIPTION
**Pull request by parthchadha**
This PR adds the following:
1. `arange`: allows start, stop or size  (or all) to be tensors.
2. Adds `floordiv` operation.
3. Allows fill operation to accept value as input.

### CODE REVIEW

File: tripy/tripy/frontend/ops/tensor_initializers.py [hunk 1]
```diff
@@ -330,21 +333,32 @@ def arange(
...
             "Arange tensor is empty.",
             details=[
                 f"start={start}, stop={stop}, step={step}",
             ],
         )
+
+    if isinstance(size, Tensor):
+        from tripy.frontend.trace.ops.cast import cast
+
+        size = cast(size, int32)
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/ops/tensor_initializers.py:358]: Can't we express this via dtype constraints?

**Review by yizhuoz004 — APPROVED** [review]

File: tripy/tripy/function_registry.py [hunk 2]
```diff
@@ -102,14 +104,23 @@ def matches_type(name: str, annotation: type, arg: Any):
             # In cases where a type is not available at the time of function definition, the type
             # annotation may be provided as a string. Since we need the actual type, we just
             # eval it here.
-            if isinstance(annotation, str):
+            import importlib
+
+            if isinstance(annotation, (str, ForwardRef)):
```
Comment (pranavm-nvidia) [inline tripy/tripy/function_registry.py:109]: @slyubomirsky also implemented this in his PR: https://github.com/NVIDIA/TensorRT-Incubator/pull/185
Which one should we keep?

Comment (parthchadha) [reply tripy/tripy/function_registry.py:109]: I can use @slyubomirsky's implementation since it implements broader support.

Comment (slyubomirsky) [reply tripy/tripy/function_registry.py:109]: Just curious, where did `ForwardRef` come into play here? I did not know about `__forward_arg__`, so I think I could use this to handle tensor literals without a workaround.

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py [hunk 3]
```diff
@@ -432,6 +432,74 @@ def __rtruediv__(self: numbers.Number, other: Union["tripy.Tensor", Any]) -> "tr
...
+@constraints.dtype_info(
+    dtype_variables={"T1": ["float32", "float16", "bfloat16", "int8", "int32", "int64"]},
+    dtype_constraints={"self": "T1", "other": "T1", constraints.RETURN_VALUE: "T1"},
+)
+def __floordiv__(self: Union["tripy.Tensor", Any], other: Union["tripy.Tensor", Any]) -> "tripy.Tensor":
+    """
+    Performs an elementwise floor division.
+
+    Args:
+        self: Tensor to be divided by other.
```
Comment (yizhuoz004) [inline tripy/tripy/frontend/trace/ops/binary_elementwise.py:446]: Need to mention "floor division"

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py [hunk 4]
```diff
@@ -432,6 +432,74 @@ def __rtruediv__(self: numbers.Number, other: Union["tripy.Tensor", Any]) -> "tr
...
+        a = tp.Tensor([4.0, 6.0])
+        b = tp.Tensor([3.0, 4.0])
+        output = a // b
+
+        assert np.array_equal(cp.from_dlpack(output).get(), np.array([1.0, 1.0]))
+    """
+    from tripy.frontend.trace.ops.cast import cast
+    from tripy.common.datatype import int32
+
+    return cast(cast(BinaryElementwise.build([self, other], BinaryElementwise.Kind.DIV), int32), self.dtype)
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/trace/ops/binary_elementwise.py:466]: Is there no floordiv stableHLO op? Casting won't work for values outside the int32 range, which I think could happen quite easily

Comment (yizhuoz004) [reply tripy/tripy/frontend/trace/ops/binary_elementwise.py:466]: Yeah, there's no floordiv in stablehlo, I only found "div + floor". Besides we better add a rewriter in MLIR-TRT that matches `div + floor` -> TRT elementwise layer `kFLOOR_DIV`. 
Still easiest approach is TRT dialect.

File: tripy/tripy/frontend/ops/tensor_initializers.py (hunk 1 above)
Comment (parthchadha) [reply tripy/tripy/frontend/ops/tensor_initializers.py:357]: size is computed from start, stop and step which are inputs to this function.  (`size = 0 - ((start - stop) // step)`)

Comment (pranavm-nvidia) [reply tripy/tripy/frontend/ops/tensor_initializers.py:343]: But if we require start/stop/step to be `int32` then we won't need to cast here right?

Comment (parthchadha) [reply tripy/tripy/frontend/ops/tensor_initializers.py:343]: These can be of float type, ex: `tp.arange(2.3, 0.8, -0.2)`

File: tripy/tripy/frontend/trace/ops/binary_elementwise.py (hunk 4 above)
Comment (parthchadha) [reply tripy/tripy/frontend/trace/ops/binary_elementwise.py:475]: I have switched to div + floor but this caused a segfault in mlir-tensorrt: https://github.com/NVIDIA/TensorRT-Incubator/issues/208. Will use the cast approach until segfault is fixed in mlir-tensorrt.

File: tripy/tripy/frontend/ops/tensor_initializers.py [hunk 5]
```diff
@@ -291,7 +290,10 @@ def triu(tensor: "tripy.Tensor", diagonal: int = 0) -> "tripy.Tensor":
     dtype_constraints={"dtype": "T1", constraints.RETURN_VALUE: "T1"},
 )
 def arange(
-    start: numbers.Number, stop: numbers.Number, step: numbers.Number = 1, dtype: "tripy.dtype" = datatype.float32
+    start: Union[numbers.Number, "tripy.Tensor"],
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/ops/tensor_initializers.py:293]: Should we mention that these need to be scalars?

**Pull request merged**